nixio/data_array.py

```diff
@@ -82,12 +82,22 @@ def append_set_dimension(self, labels=None):
         Append a new SetDimension to the list of existing dimension
         descriptors.
 
+        :param labels: The set of sample labels
+        :type labels: list or convertible to list
+
         :returns: The newly created SetDimension.
         :rtype: nixio.SetDimension
         """
         index = len(self.dimensions) + 1
         setdim = SetDimension.create_new(self, index)
-        if labels:
+        if labels is not None:
+            if not hasattr(labels, '__iter__') or isinstance(labels, str):
+                raise ValueError('`labels` has to be a list-like object.')
+            for label in labels:
+                if not isinstance(label, str):
+                    raise ValueError(f'`labels` has to contain string objects, not {type(label)}')
+            if not isinstance(labels, list):
+                labels = list(labels)
```

Comment:
I think it would make more sense to move these checks to the `labels` property setter in the `SetDimension` so that the same checks are applied when doing `set_dim.labels = ...`